modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/AbstractEventEndpoint.java

```diff
@@ -2270,7 +2270,9 @@ private URI signUrl(URI url) {
   private List<Field> getEventMediaPackageElementFields(MediaPackageElement element) {
     List<Field> fields = new ArrayList<>();
     fields.add(f("id", v(element.getIdentifier(), BLANK)));
-    fields.add(f("type", v(element.getFlavor().toString(), BLANK)));
+    if (element.getFlavor() != null) {
+      fields.add(f("type", v(element.getFlavor().toString(), BLANK)));
```

Comment:
This completely drops the type field if the flavour is unset.  Shouldn't we add an else clause that sets the value to BLANK?  The second argument to v() is what to set in the case the value is null...